cyclonedx_py/_internal/environment.py

```diff
@@ -47,6 +50,8 @@
 
     T_AllComponents = Dict[str, Tuple['Component', Iterable[Requirement]]]
 
+LICENSE_FILE_REGEX = re.compile('LICEN[CS]E.*|COPYING.*')
```

Comment:
files to consider: 
- `LICENCE`/`LICENSE` - obviously
- `NOTICE` - some license texts, like Apache-2.0, explicitly state this exact file (name) as a optional source of custom license addendum
- `COPYING` - obviously

read also: https://wheel.readthedocs.io/en/stable/user_guide.html?#including-license-files-in-the-generated-wheel-file
They describe this already, and they also allow arbitrary file suffixes.

Reply:
Ah nice catch! I think I got the original idea for the pattern from another license finder.

I'm thinking with `AUTHORS`  and `NOTICE` it's a bit less clear if some of the results should be considered license files in the sense that I'd expect in a CycloneDX SBOM, but perhaps it still makes sense. See e.g.
https://github.com/CycloneDX/cyclonedx-python/blob/main/NOTICE (maybe it counts.. not sure) 
https://github.com/python-gitlab/python-gitlab/blob/main/AUTHORS

Suffixes should already be covered but I'll make sure to add some tests for this!

Reply:
regarding `NOTICE` file:
such a file is a case of an addendum to standard license texts.
example: 
the https://github.com/CycloneDX/cyclonedx-python/blob/main/NOTICE counts, since it is anchored in the apache-2.0 license - see https://github.com/CycloneDX/cyclonedx-python/blob/main/LICENSE#L106-L121

the `AUTHORS` is unclear, this is true.

Reply:
@jkowalleck thanks! Ok makes sense. WDYT should we skip `AUTHORS` or add it but make the pattern configurable? (this would make more sense if there was a tiny high-level public API for programmatic usage perhaps, I'll open a separate issue to discuss).

Reply:
skip authors for now. if it is needed, somebody can add the feature later.